### PR TITLE
Add generated rest client for health API only

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: client-generation.ps1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+node -v
+if(! $? ){
+	throw "To use Autorest to generate REST client, Node.js is required. Version 10.15.x LTS is preferred. "
+}
+autorest --info  | Out-Null
+if(-not $? ){
+	npm install -g autorest
+}
+autorest --csharp  --v3 swagger/readme.md
```
Comment (wanlwanl) [inline client-generation.ps1:9]: please add to https://github.com/Azure/azure-signalr/blob/dev/docs/build-source.md later.

File: src/Common/REST/Generated/code-model-v1 [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        "x-ms-metadata": {
+         "apiVersions": [
+          "v1"
+         ],
+         "filename": [
+          "mem:///109?oai3.shaken.json"
+         ],
+         "name": "paths·a2exod·api-v1-health·head·responses·200",
+         "originalLocations": [
+          "file:///C:/Users/zityang/source/repos/azure-signalr/src/Common/REST/health.json#/components/responses/paths·a2exod·api-v1-health·head·responses·200"
```
Comment (wanlwanl) [inline src/Common/REST/Generated/code-model-v1:42]: Does it matters? I can see your alias

**Review by wanlwanl — APPROVED** [review]

File: src/Common/REST/Generated/code-model-v1 (hunk 2 above)
Comment (wanlwanl) [reply src/Common/REST/Generated/code-model-v1:42]: Can it be relative path?

Comment (Y-Sindo) [reply src/Common/REST/Generated/code-model-v1]: I haven't found settings about this file, and I find that there are no these file in azure-sdk-for-net repo, so I just deleted it.

File: client-generation.ps1 (hunk 1 above)
Comment (Y-Sindo) [reply client-generation.ps1:9]: As the generated file are already in the repo, they don't need to generate themselves.

File: src/Common/REST/Generated/GeneratedRestClient.cs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+{
+    using Microsoft.Rest;
+    using Microsoft.Rest.Serialization;
+    using Newtonsoft.Json;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+
+    public partial class GeneratedRestClient : ServiceClient<GeneratedRestClient>, IGeneratedRestClient
```
Comment (vicancy) [inline src/Common/REST/Generated/GeneratedRestClient.cs:21]: is the auto-generated class naming customizable?

File: src/Common/REST/Generated/GeneratedRestClient.cs [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+        /// An optional partial-method to perform custom initialization.
+        ///</summary>
+        partial void CustomInitialize();
+        /// <summary>
+        /// Initializes client properties.
+        /// </summary>
+        private void Initialize()
+        {
+            HealthApi = new HealthApi(this);
+            BaseUri = new System.Uri("http://localhost");
```
Comment (vicancy) [inline src/Common/REST/Generated/GeneratedRestClient.cs:290]: what is this for?

File: swagger/readme.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+input-file: ../src/Common/REST/health.json
+```
+
+## Alternate settings
+
+``` yaml $(csharp)
+license-header: MICROSOFT_MIT_NO_VERSION
+override-client-name: GeneratedRestClient
+namespace: Microsoft.Azure.SignalR
+output-folder: ../src/Common/REST/Generated
```
Comment (vicancy) [inline swagger/readme.md:17]: put all these files into folder `src\Microsoft.Azure.SignalR.Common\RestClients`?

File: src/Common/REST/Generated/GeneratedRestClient.cs (hunk 3 above)
Comment (Y-Sindo) [reply src/Common/REST/Generated/GeneratedRestClient.cs:21]: yes

File: src/Common/REST/Generated/GeneratedRestClient.cs (hunk 4 above)
Comment (Y-Sindo) [reply src/Common/REST/Generated/GeneratedRestClient.cs:290]: `Initialize()` is called by each constructor. This `BaseUri = new System.Uri("http://localhost");`  setting would be covered afterwards if we pass `baseUri `parameter to the constructors. For example, see [this](https://github.com/Azure/azure-signalr/blob/8b7bece7da1c98b012e0c41ceb145901ca78c10b/src/Common/REST/Generated/GeneratedRestClient.cs#L272).

**Review by vicancy — APPROVED** [review]

**Pull request merged**